### PR TITLE
Improve VectorDataExchange::Full::reset_ghost_values

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: source/matrix_free/vector_data_exchange.cc [hunk 1]
```diff
@@ -1344,10 +1344,18 @@ namespace internal
       void
       Full::reset_ghost_values_impl(const ArrayView<Number> &ghost_array) const
       {
-        // TODO
-        std::memset(ghost_array.data(),
-                    0.0,
-                    ghost_array.size() * sizeof(Number));
+        // reset ghost values coming from shared-memory neighbors
+        // TODO: only needed if values are buffered
```
Comment (peterrum) [inline source/matrix_free/vector_data_exchange.cc:1348]: I am still not completely sure where this logic should be.

File: source/matrix_free/vector_data_exchange.cc [hunk 2]
```diff
@@ -1344,10 +1344,18 @@ namespace internal
...
+        for (const auto &i : sm_export_data_this.second)
+          std::memset(ghost_array.data() + (i.first - n_local_elements),
+                      0.0,
+                      sizeof(Number) * i.second);
+
+        // reset ghost values coming from remote neighbors
+        for (const auto &i : ghost_indices_subset_data.second)
+          std::memset(ghost_array.data() + i.first,
+                      0.0,
+                      sizeof(Number) * i.second);
```
Comment (drwells) [inline source/matrix_free/vector_data_exchange.cc:1358]: Could we use `std::fill` here?

File: source/matrix_free/vector_data_exchange.cc [hunk 3]
```diff
@@ -1344,10 +1344,18 @@ namespace internal
...
       {
-        // TODO
-        std::memset(ghost_array.data(),
-                    0.0,
-                    ghost_array.size() * sizeof(Number));
+        // reset ghost values coming from shared-memory neighbors
+        // TODO: only needed if values are buffered
+        for (const auto &i : sm_export_data_this.second)
+          std::memset(ghost_array.data() + (i.first - n_local_elements),
+                      0.0,
```
Comment (kronbichler) [inline source/matrix_free/vector_data_exchange.cc:1351]: `memset` takes `int` arguments.
```suggestion
                      0,
```

Comment (peterrum) [reply source/matrix_free/vector_data_exchange.cc]: I have made the change.

File: source/matrix_free/vector_data_exchange.cc (hunk 2 above)
Comment (peterrum) [reply source/matrix_free/vector_data_exchange.cc:1358]: I left the `std::memset` because it is how it is done in `Partitioner`.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**